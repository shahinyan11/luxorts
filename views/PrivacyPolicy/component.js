import React from 'react';
import { FormattedMessage } from 'react-intl';

import LandingPagesLayout from 'views/layouts/LandingPages';

import useContainer, { getInitialProps } from './hook';

const PrivacyPolicy = () => {
  const { htmlString, titles } = useContainer();

  return (
    <section className="terms">
      <div className="terms__container">
        <div className="terms__content">
          <div className="terms__column" dangerouslySetInnerHTML={{ __html: htmlString }} />
          <div className="terms__navigation">
            <h3 className="terms__item-title mb-24 mt-76">
              <FormattedMessage id="shared.tableOfContents" />
            </h3>
            <div className="terms__group">
              <ol className="terms__anchor-list">
                {titles.map(({ id, text }, index) => (
                  <li key={id} className="terms__anchor-item">
                    <a href={`#${id}`} className="terms__anchor-link">
                      {`${index + 1}. ${text}`}
                    </a>
                  </li>
                ))}
              </ol>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

PrivacyPolicy.getInitialProps = getInitialProps;

PrivacyPolicy.Layout = LandingPagesLayout;

export default PrivacyPolicy;
