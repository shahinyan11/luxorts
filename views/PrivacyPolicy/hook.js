import { useSelector } from 'react-redux';

import buildTitles from 'utils/buildTitles';

import { fetchPrivacyPolicyPageData } from 'state/concepts/publicInfo/actions';
import { privacyPolicySelector } from 'state/concepts/publicInfo/selectors';

function useContainer() {
  const { content } = useSelector(privacyPolicySelector);
  const { titles, htmlString } = buildTitles(content);

  return {
    titles,
    htmlString,
  };
}

export const getInitialProps = async (ctx) => {
  ctx.store.dispatch(fetchPrivacyPolicyPageData());
};

export default useContainer;
