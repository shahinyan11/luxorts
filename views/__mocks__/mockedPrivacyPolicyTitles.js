export default [{ text: 'h2 text', id: 'h2_id' }];
