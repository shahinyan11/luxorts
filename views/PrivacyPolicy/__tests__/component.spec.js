import { shallow } from 'enzyme';

import mockedPrivacyPolicy from 'views/__mocks__/mockedPrivacyPolicy';
import mockedPrivacyPolicyTitles from 'views/__mocks__/mockedPrivacyPolicyTitles';

import PrivacyPolicy from '../component';

jest.mock('state/concepts/publicInfo/selectors', () => ({
  privacyPolicySelector: jest.fn(() => mockedPrivacyPolicy),
}));
jest.mock('utils/buildTitles', () =>
  jest.fn(() => ({
    titles: mockedPrivacyPolicyTitles,
    htmlString: 'htmlString',
  })),
);

describe('PrivacyPolicy component', () => {
  const component = shallow(<PrivacyPolicy />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
