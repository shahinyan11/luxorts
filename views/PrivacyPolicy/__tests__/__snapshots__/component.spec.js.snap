// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPolicy component matches snapshot 1`] = `
<section
  className="terms"
>
  <div
    className="terms__container"
  >
    <div
      className="terms__content"
    >
      <div
        className="terms__column"
        dangerouslySetInnerHTML={
          Object {
            "__html": "htmlString",
          }
        }
      />
      <div
        className="terms__navigation"
      >
        <h3
          className="terms__item-title mb-24 mt-76"
        >
          <Memo(FormattedMessage)
            id="shared.tableOfContents"
          />
        </h3>
        <div
          className="terms__group"
        >
          <ol
            className="terms__anchor-list"
          >
            <li
              className="terms__anchor-item"
              key="h2_id"
            >
              <a
                className="terms__anchor-link"
                href="#h2_id"
              >
                1. h2 text
              </a>
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
</section>
`;
